Apromore-ProcessMining-Collection/prom-bpmn/src/test/java/org/apromore/processmining/models/graphbased/directed/bpmn/ModelCheckerTest.java

```diff
@@ -0,0 +1,125 @@
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.apromore.processmining.models.graphbased.directed.bpmn;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ModelCheckerTest {
+
+    @Test
+    public void testValidModel() throws Exception {
+        BPMNDiagram diagram = getDiagramFromFile("d1_valid.bpmn");
+        ModelCheckResult modelCheckResult = ModelChecker.MODEL_CHECKER_NO_POOLS.checkModel(diagram);
+        assertTrue(modelCheckResult.isValid());
+        assertTrue(modelCheckResult.getErrorCodes().isEmpty());
+        assertEquals("", modelCheckResult.invalidMessage());
+    }
+
+    @Test
+    public void testEmptyModel() throws Exception {
+        BPMNDiagram diagram = getDiagramFromFile("empty_model.bpmn");
+        ModelCheckResult modelCheckResult = ModelChecker.MODEL_CHECKER_NO_POOLS.checkModel(diagram);
+        assertFalse(modelCheckResult.isValid());
+        assertTrue(modelCheckResult.getErrorCodes().contains(ModelCheckResult.EMPTY_MODEL_CODE));
+    }
+
+    @Test
+    public void testModelWithPool() throws Exception {
+        BPMNDiagram diagram = getDiagramFromFile("model_with_pool.bpmn");
+        ModelCheckResult modelCheckResult = ModelChecker.MODEL_CHECKER_ALLOW_POOLS.checkModel(diagram);
+        assertTrue(modelCheckResult.isValid());
+        assertTrue(modelCheckResult.getErrorCodes().isEmpty());
+        assertEquals("", modelCheckResult.invalidMessage());
+
+        modelCheckResult = ModelChecker.MODEL_CHECKER_NO_POOLS.checkModel(diagram);
+        assertFalse(modelCheckResult.isValid());
+        assertTrue(modelCheckResult.getErrorCodes().contains(ModelCheckResult.POOLS_NOT_SUPPORTED_CODE));
+    }
+
+    @Test
+    public void testModelWithSelfLoopAndMultipleEventArcs() throws Exception {
+        BPMNDiagram diagram = getDiagramFromFile("self_loop.bpmn");
+
+        ModelCheckResult modelCheckResult = ModelChecker.MODEL_CHECKER_NO_POOLS.checkModel(diagram);
+        assertFalse(modelCheckResult.isValid());
+
+        Set<Integer> errorCodes = modelCheckResult.getErrorCodes();
+        assertTrue(errorCodes.contains(ModelCheckResult.SELF_LOOP_CODE));
+        assertTrue(errorCodes.contains(ModelCheckResult.TASK_MULTIPLE_OUTGOING_ARCS_CODE));
+        assertTrue(errorCodes.contains(ModelCheckResult.TASK_MULTIPLE_INCOMING_ARCS_CODE));
+        assertTrue(errorCodes.contains(ModelCheckResult.START_MULTIPLE_OUTGOING_ARCS_CODE));
+        assertTrue(errorCodes.contains(ModelCheckResult.END_MULTIPLE_INCOMING_ARCS_CODE));
```

Comment:
Maybe it's better to provide only a method called: boolean ModelCheckResult.contains(errorCode), because the use of getErrorCodes so far is only to call its contain(). There could be a future need to get all error codes but we don't know yet, so it's usually a good practice to only provide a contains() method.